src/Post.php

```diff
@@ -75,8 +95,12 @@ private function get_query_filter_name() {
 	 */
 	public function endpoint_args() {
 		return [
+			'id' => [
+				'validate_callback' => function ( $id ) {
+					return ctype_digit( (string) $id ) && 0 !== (int) $id;
```

Comment:
But that allows '0'. I think it's best like this. It also doesn't allow you to use -66 and interpret it as 66 (which absint would).